### PR TITLE
Added block deduplication via Hazelcast for multiple instance deployment OBP-111

### PR DESCRIPTION
**Pull request by rkharisov**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -21,7 +21,11 @@ Chain adapter uses `chain-adapter.properties` as a default configuration file th
 - `chain-adapter.iroha` - Iroha host and port configuration
 - `chain-adapter.dropLastReadBlock` - as it was mentioned before, chain adapter saves the last read block height. It's possible to drop height to zero on a service startup by setting this value to `true`. Good for testing purposes. 
 - `chain-adapter.irohaCredential` - credentials of the account that will be used by the service to listen to Iroha blocks. The account must have `can_get_blocks` permission.
+- `chain-adapter.clusterEnabled` - `true` if expected to run multiple instances of the service and `false` otherwise. If `true` Iroha blocks will be deduplicated. Default value is `false`.
+- `chain-adapter.clusterHazelcastMembers` - comma separated list of Hazelcast's node addresses in format `host1:port1,host2:port2`. Required only if `chain-adapter.clusterEnabled` is `true`.
 
+=false
```
Comment (kgusman) [inline README.md]: Is it a mistake with formatting or something else?

**Review by kgusman — APPROVED** [review]

File: README.md (hunk 1 above)
Comment (rkharisov) [reply README.md]: Yes, this is mistake. Deleted it. Thank you

File: chain-adapter/src/main/kotlin/com/d3/chainadapter/dedup/hazelcast/HazelcastBlockProcessor.kt [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+/**
+ * This implementation sends new block to RMQ only if it was not processed before by this or another
+ * instance of service. It allows to run multiple instances of the service. Deduplication provided
+ * by pessimistic lock on distributed map.
+ */
+class HazelcastBlockProcessor(private val hazelcastInstance: HazelcastInstance) : BlockProcessor, Closeable {
+
+    private val processedBlockNumber : IMap<String, Long> = hazelcastInstance.getMap(PROCESSED_BLOCK_ACCESS_KEY)
+
+    override fun onNewBlock(block: Block, publish: (block: Block) -> Unit) {
```
Comment (Mingela) [inline chain-adapter/src/main/kotlin/com/d3/chainadapter/dedup/hazelcast/HazelcastBlockProcessor.kt:35]: it is named `action` in the other implementation and `publish` here, could you choose the same name please?

**Review by Mingela — APPROVED** [review]

File: chain-adapter/src/main/kotlin/com/d3/chainadapter/dedup/hazelcast/HazelcastBlockProcessor.kt (hunk 2 above)
Comment (rkharisov) [reply chain-adapter/src/main/kotlin/com/d3/chainadapter/dedup/hazelcast/HazelcastBlockProcessor.kt:35]: Fixed

**Pull request merged**